lib/ansible/vars/manager.py

```diff
@@ -206,20 +206,18 @@ def get_vars(self, play=None, host=None, task=None, include_hostvars=True, inclu
 
             def _get_plugin_vars(plugin, path, entities):
                 data = {}
-                try:
+                if hasattr(plugin, 'get_vars'):
                     data = plugin.get_vars(self._loader, path, entities)
-                except AttributeError:
-                    try:
-                        for entity in entities:
-                            if isinstance(entity, Host):
-                                data.update(plugin.get_host_vars(entity.name))
-                            else:
-                                data.update(plugin.get_group_vars(entity.name))
-                    except AttributeError:
-                        if hasattr(plugin, 'run'):
-                            raise AnsibleError("Cannot use v1 type vars plugin %s from %s" % (plugin._load_name, plugin._original_path))
+                elif hasattr(plugin, 'get_host_vars') and hasattr(plugin, 'get_group_vars'):
+                    for entity in entities:
+                        if isinstance(entity, Host):
+                            data.update(plugin.get_host_vars(entity.name))
                         else:
-                            raise AnsibleError("Invalid vars plugin %s from %s" % (plugin._load_name, plugin._original_path))
+                            data.update(plugin.get_group_vars(entity.name))
+                elif hasattr(plugin, 'run'):
+                    raise AnsibleError("Cannot use v1 type vars plugin %s from %s" % (plugin._load_name, plugin._original_path))
+                else:
```

Comment:
You can pass the traceback for -vvv and loggign by adding `orig_exc=e` to the AnsibleError (assuming `as e` at `Except`)